cogboard-webapp/src/components/AddEndpoint.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+
+import { useToggle } from '../hooks';
+import { saveEndpoint } from '../actions/thunks';
+
+import { IconButton, Button } from '@material-ui/core';
+import { Add } from '@material-ui/icons';
+import AppDialog from './AppDialog';
+import EndpointForm from './EndpointForm';
+
+const AddEndpoint = ({ largeButton, dataChanged, endpointsData }) => {
+  const [dialogOpened, openDialog, handleDialogClose] = useToggle();
+  const dispatch = useDispatch();
+
+  const handleAddEndpointClick = event => {
+    event.stopPropagation();
+    openDialog();
+  };
+
+  const handleSubmit = values => {
+    dispatch(saveEndpoint(values));
+    if (dataChanged !== undefined) {
+      dataChanged();
+    }
+    handleDialogClose();
+  };
+
+  return (
+    <>
+      {largeButton ? (
```

Comment:
Same thing here, you even duplicate the code. Maybe extract to a separate component?